yesod-core/Yesod/Core/FormHelpers.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Yesod.Core.FormHelpers
+    (
+      HttpMethod(..),
+      FormStyle,
+      httpLink,
+      buttonLink,
+      link
+    ) where
+
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Trans.Control (MonadBaseControl (..))
+import Yesod.Routes.Class
+import Control.Monad.Trans.Resource (MonadResource (..), MonadThrow (..))
+import Yesod.Core.Widget
+import Yesod.Core.Handler
+import Text.Blaze (ToMarkup)
+import Data.String (IsString)
+import Text.Lucius
+
+data HttpMethod = HttpGET | HttpPOST | HttpPATCH | HttpUPDATE | HttpDELETE
```

Comment:
What value does this add instead of just using our more standard textual representation of a method? The HTTP spec leaves the number of HTTP methods open-ended. And this list already contains at least one notable omission: `PUT`.